src/dg/dg.cpp

```diff
@@ -622,28 +642,138 @@ void DGBase<dim,real,MeshType>::assemble_cell_residual (
+//                    if (cell_index == 0 && iface == 0)
+//                    {
+//                        fe_values_collection_face_int.reinit(current_cell, iface, i_quad, i_mapp, i_fele);
+//                        neighbor_cell = dof_handler.begin_active();
+//                       for (unsigned int i = 0 ; i < triangulation->n_active_cells() - 1; ++i)
+//                       {
+//                           ++neighbor_cell;
+//                       }
+//                        neighbor_dofs_indices.resize(n_dofs_curr_cell);
+//                        neighbor_cell->get_dof_indices(neighbor_dofs_indices);
+//                         const unsigned int fe_index_neigh_cell = neighbor_cell->active_fe_index();
+//                        const unsigned int quad_index_neigh_cell = fe_index_neigh_cell;
+//                        const unsigned int mapping_index_neigh_cell = 0;
+//
+//                        fe_values_collection_face_ext.reinit(neighbor_cell,(iface == 1) ? 0 : 1,quad_index_neigh_cell,mapping_index_neigh_cell,fe_index_neigh_cell);
+//
+//                    }
+                   // else if (cell_index == (int) triangulation->n_active_cells() - 1 && iface == 1)
+                    if (cell_index == (int) triangulation->n_active_cells() - 1 && iface == 1)
+                    {
+                     //   fe_values_collection_face_int.reinit(current_cell, iface, i_quad, i_mapp, i_fele);
+                        neighbor_cell = dof_handler.begin_active();
+                        neighbor_dofs_indices.resize(n_dofs_curr_cell);
+                        neighbor_cell->get_dof_indices(neighbor_dofs_indices);
+                        const unsigned int fe_index_neigh_cell = neighbor_cell->active_fe_index();
+                        const unsigned int quad_index_neigh_cell = fe_index_neigh_cell;
+                        const unsigned int mapping_index_neigh_cell = 0;
+                        fe_values_collection_face_ext.reinit(neighbor_cell,(iface == 1) ? 0 : 1, quad_index_neigh_cell, mapping_index_neigh_cell, fe_index_neigh_cell); //not sure how changing the face number would work in dim!=1-dimensions.
+                    }
+                    else {
+                        continue;
+                    }
+
+                    //std::cout << "cell " << current_cell->index() << "'s " << iface << "th face has neighbour: " << neighbor_cell->index() << std::endl;
+                    const int neighbor_face_no = (iface ==1) ? 0:1;
+                    const unsigned int fe_index_neigh_cell = neighbor_cell->active_fe_index();
+
+                    //check neighbour cell face on boundary
+                    auto neigh_face_check = neighbor_cell->face(neighbor_face_no);
+                    if(neigh_face_check->at_boundary()){
+                        //do nothing
+                    }
+                    else{
+                        pcout<<"FACE NOT ON BOUNDARY LOL"<<std::endl;
```

Comment:
Improve this print statement, ideally we don't want LOLs in the code

Reply:
fixed